File: LCA/Java/LCA.java
import java.util.ArrayList;
import java.util.List;
 

class Node {
    int data;
    Node left, right;
 
    Node(int value) {
        data = value;
        left = right = null;
    }
}

public class LCA
{

    Node root;
    private List<Integer> path1 = new ArrayList<>();
    private List<Integer> path2 = new ArrayList<>();
 
    int findLCA(int n1, int n2) {
    }
 
    private int findLCAInternal(Node root, int n1, int n2) {
    }
     
    private boolean findPath(Node root, int n, List<Integer> path)
    {
    }
    public static void main(String[] args)
    {
    }
}

File: LCA/Python/LCA.py

class Node:
    
    def __init__(self, key):
        self.key =  key
        self.left = None
        self.right = None
 

def findPath( root, path, k):
    if root is None:
        return False

    path.append(root.key)

    if root.key == k :
        return True

    if ((root.left != None and findPath(root.left, path, k)) or
            (root.right!= None and findPath(root.right, path, k))):
        return True

    path.pop()
    return False

def findLCA(root, n1, n2):
